frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx

```diff
@@ -75,48 +64,83 @@ const VMRow = ({ obj: vm }: React.ComponentProps<typeof ResourceRow>) => {
+        )}
+      </div>
       {/* TODO(mlibra): migrate actions in a follow-up
       <div className="dropdown-kebab-pf">
       <ResourceKebab actions={menuActions} kind={VirtualMachineModel.kind} resource={vm} resources={[
-        getResource(VirtualMachineInstanceModel, { name, namespace, isList: false }),
-        getResource(VirtualMachineInstanceMigrationModel, {namespace}),
-        getResource(PodModel, {namespace}),
+      getResource(VirtualMachineInstanceModel, { name, namespace, isList: false }),
+      getResource(VirtualMachineInstanceMigrationModel, {namespace}),
+      getResource(PodModel, {namespace}),
       ]} />
     </div>
     */}
     </ResourceRow>
   );
 };
 
-const VMList: React.FC = (props) => <List {...props} Header={VMHeader} Row={VMRow} />;
-VMList.displayName = 'VMList';
+const VmRow = (props: VmRowProps) => {
+  const { loadError, loaded } = props;
 
-const filters = undefined;
-/* TODO(mlibra): introduce extension point for list.tsx and reenable here then.
-Beyond recent kubevirt-web-ui functionality, user experience would be improved by full-featured VM status filtering.
-To properly determine VM status, aditional objects (pods or CRs) needs to be accessed.
-Recent filtering logic around list.tsx can handle just a single "listed" object (in our case, a VirtualMachine).
-We will need to find a way how to supply additional resources there.
-
-const filters = [{
-  type: 'vm-status',
-  selected: VM_SIMPLE_STATUS_ALL,
-  reducer: getSimpleVmStatus,
-  items: VM_SIMPLE_STATUS_ALL.map(status => ({ id: status, title: VM_SIMPLE_STATUS_TO_TEXT[status] }) ),
-}];
-*/
+  if (loadError) {
+    if (loaded && loadError instanceof TimeoutError) {
+      // eslint-disable-next-line no-console
+      console.info('ResourceRow retrieval timed out.');
+      return <VmRowDetail {...props} />;
+    }
 
-type VirtualMachinesPageProps = {
-  namespace: string;
+    // eslint-disable-next-line no-console
+    console.warn('Error loading ResourceRow: ', loadError.message);
+    return <VmRowDetail loadErrorMsg="Loading error" {...props} />;
+  }
+
+  return <VmRowDetail {...props} />;
 };
 
+const VMRowFirehose = ({ obj: vm }: React.ComponentProps<typeof ResourceRow>) => {
+  const name = getName(vm);
+  const namespace = getNamespace(vm);
+
+  const vmiRes = getResource(VirtualMachineInstanceModel, {
+    name,
+    namespace,
+    isList: false,
+    props: 'vmi',
+    optional: true,
+  });
+
+  const resources = [
+    getResource(PodModel, { namespace, prop: 'pods' }),
+    getResource(VirtualMachineInstanceMigrationModel, { namespace, prop: 'migrations' }),
+    vmiRes,
+  ];
+
+  return (
+    <Firehose resources={resources}>
+      <VmRow vm={vm} />
+    </Firehose>
+  );
+};
+
+const VMList: React.FC = (props) => <List {...props} Header={VMHeader} Row={VMRowFirehose} />;
+VMList.displayName = 'VMList';
+
+const filters = [
+  {
+    type: kubevirtTableFilters.VmStatus.type,
```

Comment:
Wouldn't it be simpler to define whole `vmStatusFilter` object in `kubevirtTableFilters`, export it and just pass it in rowFilters prop?

Reply:
See https://github.com/openshift/console/pull/1786/commits/38a3d7aef163277ca332b32694a03714b4f190a2

Reply:
good idea, thanks. I've moved `Filter` type to `@console/shared`.